react/features/mobile/audio-mode/middleware.js

```diff
@@ -53,5 +58,28 @@ MiddlewareRegistry.register(({ getState }) => next => action => {
         }
     }
 
-    return next(action);
+    return result;
 });
+
+/**
+ * Sets the audio mode after a failed conference based on the reason
+ * of the failure.
+ *
+ * @private
+ * @param {Object} state - The Redux state.
+ * @param {Object} action - The dispatched Redux action.
+ * @returns {Object | undefined}
+ */
+function _getFailedConferenceAudioMode(state: Object, action: Object) {
+    const AudioMode = NativeModules.AudioMode;
+    const { error } = action;
+
+    switch (error.name) {
```

Comment:
can this logic be based on the error.recoverable flag instead of the error.name ? this way it would handle the authentication case as well.